contracts/contracts/vault/VaultAdmin.sol

```diff
@@ -221,67 +204,12 @@ contract VaultAdmin is VaultStorage {
 
             // Call harvest after withdraw in case withdraw triggers
             // distribution of additional reward tokens (true for Compound)
-            _harvest(_addr);
-            emit StrategyRemoved(_addr);
-        }
-    }
-
-    /**
-     * @dev Add a swap token to the tokens that get liquidated for stablecoins
-     *      whenever swap is called. The token must have a valid feed registered
-     *      with the price provider.
-     * @param _addr Address of the token
-     */
-    function addSwapToken(address _addr) external onlyGovernor {
-        for (uint256 i = 0; i < swapTokens.length; i++) {
-            if (swapTokens[i] == _addr) {
-                revert("Swap token already added");
-            }
-        }
-
-        // Revert if feed does not exist
-        IOracle(priceProvider).price(_addr);
-
-        swapTokens.push(_addr);
-
-        // Give Uniswap infinte approval
-        if (uniswapAddr != address(0)) {
-            IERC20 token = IERC20(_addr);
-            token.safeApprove(uniswapAddr, 0);
-            token.safeApprove(uniswapAddr, type(uint256).max);
-        }
-
-        emit SwapTokenAdded(_addr);
-    }
 
-    /**
-     * @dev Remove a swap token from the tokens that get liquidated for stablecoins.
-     * @param _addr Address of the token
-     */
-    function removeSwapToken(address _addr) external onlyGovernor {
-        uint256 swapTokenIndex = swapTokens.length;
-        for (uint256 i = 0; i < swapTokens.length; i++) {
-            if (swapTokens[i] == _addr) {
-                swapTokenIndex = i;
-                break;
-            }
-        }
-
-        require(swapTokenIndex != swapTokens.length, "Swap token not added");
-
-        // Shift everything after the index element by 1
-        for (uint256 i = swapTokenIndex; i < swapTokens.length - 1; i++) {
-            swapTokens[i] = swapTokens[i + 1];
-        }
-        swapTokens.pop();
-
-        if (uniswapAddr != address(0)) {
-            IERC20 token = IERC20(_addr);
-            // Remove Uniswap approval
-            token.safeApprove(uniswapAddr, 0);
+            // TODO: calling `_harvest` is replaced by `harvestAndSwap`. Good for an extra pair
+            // of eyes to check this, though it shouldn't be a problem IMHO [domeng]
+            IHarvester(harvesterAddress).harvestAndSwap(_addr);
```

Comment:
@tomlinton, I'm still thinking we don't need this. We can always drive the harvestAndSwap later from the harvester contract, even if the strategy is no longer an active vault strategy. 

If we remove this auto harvest, we gain two things:
- VaultAdmin contract no longer depends on the harvester contract either by code, nor has to know its address.
- If a strategy fails in such a way that harvesting is broken, we can still remove it without first upgrading either the vault or the strategy code.

Reply:
Yep I'm fine with that

Reply:
<img width="196" alt="image" src="https://user-images.githubusercontent.com/837/150374863-bfeae88e-c71c-4e7d-9a91-ef55b4bfa203.png">

Reply:
Just one thing, I will be adding a check in harvest & swap that the `_strategyAddress` parameter is one of the approved strategies. So in theory if `harvestAndSwap` fails and we remove a strategy. Harvester will no longer be able to harvest funds.

Reply:
Removed the call to harvestAndSwap and added a comment, to be careful and call `harvestAndSwap` manually before removing the strategy. Since it is no longer possible after removal.